Lib/fontTools/ttLib/sfnt.py

```diff
@@ -478,7 +478,17 @@ class WOFFDirectoryEntry(DirectoryEntry):
 
 	format = woffDirectoryEntryFormat
 	formatSize = woffDirectoryEntrySize
-	zlibCompressionLevel = ZLIB_COMPRESSION_LEVEL
+
+	def __init__(self):
+		super(WOFFDirectoryEntry, self).__init__()
+		# With fonttools<=3.1.2, the only way to set a different zlib
+		# compression level for WOFF directory entries was to set the class
+		# attribute 'zlibCompressionLevel'. This is now replaced by a globally
+		# defined `ZLIB_COMPRESSION_LEVEL`, which is also applied when
+		# compressing the metadata. For backward compatibility, we still
+		# use the class attribute if it was already set.
+		if not hasattr(WOFFDirectoryEntry, 'zlibCompressionLevel'):
+			self.zlibCompressionLevel = ZLIB_COMPRESSION_LEVEL
```

Comment:
indeed, that's what I wanted to achieve. I wanted to set the instance's `zlibCompressionLevel` to the global `ZLIB_COMPRESSION_LEVEL`; and only use the class attribute `zlibCompressionLevel` for existing code that might still use the previous API, whereby changing the compression level was defined at the class level.
Now I defined it at module scope, so it affects both the directory entries and the metadata.